### PR TITLE
Add Faculty facet to modules

### LINKED ISSUE #25: [Design] Group by Faculty
Why the courses are no longer grouped in the new release?

The issue with this design is that I couldn't browse for the courses under a certain school.
For example: I wanted to browse for all module under "School of Business", or "School of Computing", I have to find out what are the courses from school of computing and school of business. Such as:
1. Information System
2. Computer Science
3. Financial Accounting
4. Strategy & Policy
5. etc..

screenshot from 2014-07-07 14 31 42

### PR DESCRIPTION
**Pull request by ngzhian**
Fixes #25

### CODE REVIEW

**Comment by ngzhian** [comment]
@ahbeng do you think we should remove the existing "Faculty / Department" facet, or rename it to "Department"?

**Comment by ngzhian** [comment]
@yangshun @9gix any thoughts? :)

**Comment by yangshun** [comment]
@ngzhian I’ll look at it tonight (:

**Comment by ahbeng** [comment]
I think "Faculty / Department" may have been since some smaller faculties have only one department so they are named the same? Renaming to "Department" sounds fine since there's now "Faculty".

Why do you think "Department" should  be replaced, they still filter separate things, as far as I understand?

**Comment by yangshun** [comment]
> My thought is that we should replace the existing "Faculty / Department" with this new "Faculty" one

We shouldn't do this. Filtering by course / dept is still a valuable feature to have. Arts students will appreciate this especially. FASS has over 1.3k modules and filtering out just FASS modules isn't very useful.

I think renaming "Faculty / Department" -> "Department" makes sense.

Could you also help to reorder the categories in the following order (if you think there is another order that makes more sense, feel free to comment)

- Semesters
- Types
- Faculty
- Department
- Level
- MCs
- ... (the rest remains)

File: app/scripts/modules/views/ModulesView.js [hunk 1]
```diff
@@ -40,6 +40,13 @@ module.exports = Marionette.LayoutView.extend({
 
   initialize: function (options) {
     this.mods = options.mods;
+    const departmentToFaculty = {}
```
Comment (yangshun) [inline app/scripts/modules/views/ModulesView.js]: In Safari you'll get an error: `SyntaxError: Unexpected keyword 'const'. Const declarations are not supported in strict mode.`

File: app/scripts/modules/views/ModulesView.js [hunk 2]
```diff
@@ -40,6 +40,13 @@ module.exports = Marionette.LayoutView.extend({
...
+    Object.keys(options.facultyDepartments).map((fac) => {
```
Comment (yangshun) [inline app/scripts/modules/views/ModulesView.js]: AFAIK not all browsers support es6 at the moment, so it'll break in Safari. Please refrain from using es6 syntax at the moment. We'll get there eventually >_<

File: app/scripts/modules/views/ModulesView.js [hunk 3]
```diff
@@ -40,6 +40,13 @@ module.exports = Marionette.LayoutView.extend({
...
+      options.facultyDepartments[fac].map((dept) => {
+        departmentToFaculty[dept] = fac
+      })
```
Comment (yangshun) [inline app/scripts/modules/views/ModulesView.js]: Missing `;`

File: app/scripts/modules/views/ModulesView.js [hunk 4]
```diff
@@ -40,6 +40,13 @@ module.exports = Marionette.LayoutView.extend({
 
   initialize: function (options) {
     this.mods = options.mods;
+    const departmentToFaculty = {}
+    Object.keys(options.facultyDepartments).map((fac) => {
+      options.facultyDepartments[fac].map((dept) => {
+        departmentToFaculty[dept] = fac
```
Comment (yangshun) [inline app/scripts/modules/views/ModulesView.js]: Missing `;`

File: app/scripts/modules/views/ModulesView.js [hunk 5]
```diff
@@ -40,6 +40,13 @@ module.exports = Marionette.LayoutView.extend({
...
+      })
+    })
```
Comment (yangshun) [inline app/scripts/modules/views/ModulesView.js]: Missing `;`

**Comment by yangshun** [comment]
LGTM, except for the es6 syntax and semicolons. Or should we not use semicolons? Up for debate!

**Comment by ngzhian** [comment]
Ah makes sense, sorry I was being selfish and thinking that department helps (cos that's what i care about :P)
So I'll:
1. Rename "Faculty / Department" -> "Department"
2. Reorder the facets, I think there's no correct order, and I like the one suggested, so will update it to that :)

File: app/scripts/modules/views/ModulesView.js (hunk 2 above)
Comment (ngzhian) [inline app/scripts/modules/views/ModulesView.js]: ahh shit okay, i keep assuming that browserify will compile it to es5... we need babel :P

**Comment by ngzhian** [comment]
Updated, ptal :)

**Pull request merged**